### PR TITLE
Properties case style fix

### PR DESCRIPTION
**Pull request by isaul32**
- Refactor gradle.properties file
  - snake case -> camel case (https://docs.gradle.org/current/userguide/build_environment.html#sec:gradle_configuration_properties)
  - Fix mvnrepository.com links
  - Remove unused log4j2_mock_version property

---

Please make sure the below checklist is followed for Pull Requests.

-   [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [x] Tests are added where necessary
-   [x] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/server/templates/gradle.properties.ejs [hunk 1]
```diff
@@ -20,65 +20,62 @@ rootProject.name=<%= dasherizedBaseName %>
...
-spring_boot_version=<%= SPRING_BOOT_VERSION %>
+# https://mvnrepository.com/artifact/io.github.jhipster/jhipster-dependencies/<%= JHIPSTER_DEPENDENCIES_VERSION %>
+springBootVersion=<%= SPRING_BOOT_VERSION %>
 # The hibernate version should match the one managed by
-# https://mvnrepository.com/artifact/org.springframework.boot/spring-boot-dependencies/${spring-boot.version} -->
-hibernate_version=<%= HIBERNATE_VERSION %>
-mapstruct_version=1.3.1.Final
-archunit_junit5_version=0.14.1
-<%_ if (searchEngine === 'elasticsearch') { _%>
-log4j2_mock_version=0.0.2
```
Comment (murdos) [inline generators/server/templates/gradle.properties.ejs:37]: 👍 I forgot to remove it during the Elasticsearch upgrade: https://github.com/jhipster/generator-jhipster/pull/11683
Could you do me a favour, and also remove it in `pom.xml.ejs` template?

Comment (isaul32) [reply generators/server/templates/gradle.properties.ejs:37]: Sure :slightly_smiling_face:

**Review by DanielFran — APPROVED** [review]
LGTM

**Comment by pascalgrimaud** [comment]
Thanks and congrats for your 1st contribution @isaul32 !

**Pull request merged**